csrc/domain_view.cpp

```diff
@@ -0,0 +1,138 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <domain_view.h>
+
+#include <utility>
+
+namespace nvfuser {
+
+void IterDomainLikeObjectView::split(
+    int64_t axis,
+    Val* factor,
+    bool inner_split) {
+  NVF_ERROR(false, "Not implemented yet");
+}
+
+void IterDomainLikeObjectView::split(
+    int64_t axis,
+    int64_t factor,
+    bool inner_split) {
+  return split(axis, IrBuilder::create<Val>(factor), inner_split);
+}
+
+namespace {
+
+struct DispatchMerge {
+  template <typename LHS, typename RHS>
+  IDLO operator()(LHS&& lhs, RHS&& rhs) const {
+    using L = std::decay_t<LHS>;
+    using R = std::decay_t<RHS>;
+    if constexpr (
+        std::is_same_v<L, std::monostate> ||
+        std::is_same_v<R, std::monostate>) {
+      NVF_CHECK(false, "Unsupported type in IDLOView::merge");
+      return {};
+    } else if constexpr (
+        std::is_same_v<L, IterDomain*> && std::is_same_v<R, IterDomain*>) {
+      return IterDomain::merge(std::forward<LHS>(lhs), std::forward<RHS>(rhs));
+    } else if constexpr (
+        std::is_same_v<L, ValGroupAndItsGraph> &&
+        std::is_same_v<R, ValGroupAndItsGraph>) {
+      NVF_CHECK(
+          lhs.graph == rhs.graph,
+          "Can not merge ValGroups of different graph.");
+      auto graph = lhs.graph;
+      // If there is already an existing merge in the ValGraph, just use it.
+      auto lhs_uses = graph->getUses(lhs.group);
+      for (const ExprGroup& use : lhs_uses) {
+        auto input_groups = graph->inputGroups(use);
+        NVF_ERROR(input_groups.size() == 2);
```

Comment:
Can you assume this?

Reply:
No, we can not. I have to check that it is a group of merges before checking this.

Reply:
fixed